data_validation/partition_builder.py

```diff
@@ -320,27 +341,24 @@ def _add_partition_filters(
 
             yaml_configs_list[ind] = {
                 "target_folder_name": config_manager.full_source_table,
-                "partitions": [],
+                "yaml_files": [],
             }
-            for pos in range(len(filter_list[0])):
-                filter_dict = {
-                    "type": "custom",
-                    "source": filter_list[0][pos],
-                    "target": filter_list[1][pos],
-                }
-                # Append partition new filter
-                config_manager.filters.append(filter_dict)
 
+            # Create a list of lists chunked by partitions per file
+            source_filters_list = list_to_sublists(
+                filter_list[0], self.args.parts_per_file
+            )
+            target_filters_list = list_to_sublists(
+                filter_list[1], self.args.parts_per_file
+            )
+            for i, source_filters in enumerate(source_filters_list):
```

Comment:
Why did we enumerate only source_filters_list? It's confusing because we use both source and target filters here.
It may be more simple to do:
```
for i in len(source_filters_list):
                yaml_config = self._add_filters_get_yaml(
                    config_manager, source_filters_list[i], target_filters_list[i]
                )
```